src/pymor/algorithms/lyapunov.py

```diff
@@ -7,18 +7,25 @@
 
 from pymor.core.config import config
 from pymor.core.defaults import defaults
+from pymor.tools.frozendict import FrozenDict
 from pymor.operators.interface import Operator
 
-_DEFAULT_LYAP_LRCF_SPARSE_SOLVER_BACKEND = ('pymess' if config.HAVE_PYMESS else
-                                            'lradi')
 
-_DEFAULT_LYAP_LRCF_DENSE_SOLVER_BACKEND = ('pymess' if config.HAVE_PYMESS else
-                                           'slycot' if config.HAVE_SLYCOT else
-                                           'scipy')
-
-_DEFAULT_LYAP_DENSE_SOLVER_BACKEND = ('pymess' if config.HAVE_PYMESS else
-                                      'slycot' if config.HAVE_SLYCOT else
-                                      'scipy')
+_DEFAULT_LYAP_SOLVER_BACKEND = FrozenDict(
```

Comment:
What is your reason for going with `FrozenDict` here? I'd say it's fine to use a plain dict here (and leave some room for hacking). So far, `FrozenDict` is only used to ensure immutability of immutable objects, where it is relatively likely that  changing values could cause unexpected side effects

Reply:
@pmli suggested this. I think the idea is that these are built-in defaults and different values can always be passed to the functions. I'm happy to change it though :)

Reply:
@sdrave The solver backend could already be modified via `defaults` of `solve_cont_lyap_lrcf` and other methods. Having a mutable global variable feels a bit strange to me...

Reply:
I know. I still do not see any real benefit in using `FrozenDict` here so I'd leave it out. But this is probably a matter of taste, so I would leave it up to @artpelling's taste ..

Reply:
Don't make me choose sides :cold_sweat:

To resolve this, I would stay with the `FrozenDict` as no functionality is lost and there does not yet seem to be a unified way to handle this. If we decide on a paradigm globally this should be easily adjusted.

Reply:
Fine with me ..